client/src/components/Chat.jsx

```diff
@@ -323,89 +320,145 @@ const Chat = () => {
+								}
+							})() : null
+
+							// is this message currently being replied? 
+							const hasActiveReply = currentReplyMessageId === id
 
 							return (
-								<div
-									key={id}
-									id={!isSender && `message-${id}`}
-									className={`w-full flex text-white relative mb-2 ${
-										isSender ? 'justify-end' : 'justify-start'
-									}`}
-								>
+								<div key={id} id={`message-${id}`} className={createClassesFromArray(
+									'flex flex-col gap-2 py-2 duration-500 transition-all',
+									hasActiveReply && 'bg-[#FF9F1C]/25 border-[#FF9F1C]',
+									hasActiveReply && (isSender ? 'border-r-[3.5px]' : 'border-l-[3.5px]')
+								)}>
+								{replyTo && (
 									<div
-										className={`flex flex-col mb-[2px] min-w-[10px] mdl:max-w-[80%] max-w-[50%] ${
-											isSender ? 'items-end' : 'items-start'
-										}`}
+										className={createClassesFromArray(
+											'max-w-[80%] md:max-w-[50%] min-w-[10px] flex gap-2 items-center',
+											sender.toString() === senderId.toString() && 'self-end',
+											repliedMessage && 'cursor-pointer'
+										)}
+										onClick={() => scrollToMessage(replyTo)}
 									>
-										{containsBadword && !isSender && !badwordChoices[id] ? (
-											<div className='flex flex-col border-red border w-full rounded-r-md p-3'>
-												<p>Your buddy is trying to send you a bad word</p>
-												<div className='flex w-full gap-6'>
-													<span onClick={() => showBadword(id)} className='text-sm cursor-pointer'>See</span>
-													<span onClick={() => hideBadword(id)} className='text-red text-sm cursor-pointer'>Hide</span>
-												</div>
-											</div>
-										)
-											:
-											<>
-												<div
-													className={`chat bg-red p-3 break-all will-change-auto flex gap-6 items-center text ${isSender
-														? 'justify-between bg-secondary rounded-l-md'
-														: 'rounded-r-md'
-														}`}
-												>
-													{typeof message === 'string' ? (
-														<span dangerouslySetInnerHTML={{
-															__html: md.render(
-																badwordChoices[id] === 'hide' ? badwords.filter(message) : badwordChoices[id] === 'show' ? message : message)
-														}} />
-													) : (
-														badwordChoices[id] === 'hide' ? badwords.filter(message) : badwordChoices[id] === 'show' ? message : message
-													)}
-
-													<DropDownOptions
-														isSender={
-															isSender
-															&& status !== 'pending'}
-														id={id}
-														inputRef={inputRef}
-														cancelEdit={cancelEdit}
-														setEditing={setEditing}
-														setIsQuoteReply={setIsQuoteReply}
-														setQuoteMessage={setQuoteMessage}
+										<div className="truncate border-b-4 border-[#FF9F1C] p-1">
+											{repliedMessage ? (
+												typeof repliedMessage.message === 'string' ? (
```

Comment:
what's the logic behind here?

Reply:
Displaying a preview of the replied message

Reply:
Got ya